tests/equation_bcs/test_equation_bcs_assemble.py

```diff
@@ -15,8 +15,6 @@ def test_equation_bcs_direct_assemble_one_form():
 
     g = assemble(F, bcs=bc.extract_form('F'))
     assert np.allclose(g.dat.data, [0.5, 0.5, 0, 0])
-    g = assemble(F, bcs=bc)
-    assert np.allclose(g.dat.data, [0.5, 0.5, 0, 0])
```

Comment:
Why are you removing this test?

Reply:
This test does the same thing as the one right above, but it uses a shortcut that is only valid for 1-forms. I removed this test as I disabled the shortcut to simplify `assemble.py`.

Reply:
Simplification is always good!